server/publicapi/items/service.py

```diff
@@ -137,14 +130,39 @@ def on_fetched(self, result):
                 url_parts.query
             )
 
+    def _process_fetched_object(self, document):
+        """Does some processing on the raw document fetched from database.
+
+        It sets the item's `uri` field and removes all the fields added by the
+        `Eve` framework that are not part of the NINJS standard (except for
+        the HATEOAS `_links` object).
+        It also sets the URLs for all externally referenced media content.
+
+        :param dict document: MongoDB document to process
+        """
+        document['uri'] = self._get_uri(document)
+
+        for field_name in ('_id', '_etag', '_created', '_updated'):
+            document.pop(field_name, None)
+
+        if 'renditions' in document:
+            for k, v in document['renditions'].items():
+                if 'media' in v:
+                    v['href'] = url_for_media(v['media'])
+
     def _get_uri(self, document):
         """Return the given document's `uri`.
 
         :param dict document: MongoDB document fetched from database
         """
+        if document.get('type') == 'composite':
+            endpoint_name = 'packages'
+        else:
+            endpoint_name = 'items'
+
         resource_url = '{api_url}/{endpoint}/'.format(
             api_url=app.config['PUBLICAPI_URL'],
-            endpoint=app.config['URLS'][self.datasource]
+            endpoint=app.config['URLS'][endpoint_name]
```

Comment:
**Explanation:** `self.datasource` is not used anymore, because `PackagesService` might use this method to generate links for the _items_ it references, resulting in invalid URLs (i.e. with correct item ID, but using the `/packages` endpoint instead of  `/items`).